File: RUBY/zadanie_1.rb
def find_matching_substrings(s)
    # –егул€рное выражение дл€ соответстви€ шаблону 1[0]1, где [0] обозначает один или более нулей
    pattern = /10+1/
    # Ќайти все совпадени€ шаблона в строке
    s.scan(pattern)
end

# ѕример строки из задачи
example_str = "101tf11flb10001"
# Ќайти и вывести все совпадающие подстроки
matching_substrings = find_matching_substrings(example_str)
puts matching_substrings

File: SWIFT/zadanie_1.swift
import Foundation

func findMatchingSubstrings(s: String) -> [String] {
    // –егул€рное выражение дл€ соответстви€ шаблону 1[0]1, где [0] обозначает один или более нулей
    let pattern = "10+1"
    let regex = try! NSRegularExpression(pattern: pattern)
    let matches = regex.matches(in: s, range: NSRange(s.startIndex..., in: s))
    return matches.map {
        String(s[Range($0.range, in: s)!])
    }
}

// ѕример строки из задачи
let exampleStr = "101tf11flb10001"
// Ќайти и вывести все совпадающие подстроки
let matchingSubstrings = findMatchingSubstrings(s: exampleStr)
print(matchingSubstrings)
